### PR TITLE
CI: Fix checks fallback logic when changed files listing is missing

### PR DESCRIPTION
**Pull request by akien-mga**
Follow-up to #76263 and #76296.

The file would be one byte (newline) so the check with '-s' failed.

### CODE REVIEW

**Review by aaronfranke — APPROVED** [review]

**Comment by YuriSizov** [comment]
Note to maintainers: Don't merge this one. We need to test it with a batch merge, so it will be included into one of those.

**Comment by YuriSizov** [comment]
Cherry-picked for 4.0.3.

**Pull request merged**
